www/src/components/button.js

```diff
@@ -41,8 +42,42 @@ const Button = ({
     },
   }
 
+  const trackingOnClick = e => {
+    if (typeof props.onClick === `function`) {
+      props.onClick(e)
+    }
+
+    let redirect = true
+
+    // Slightly modified logic from the gatsby-plugin-google-analytics
+    // But this one should work with `Link` component as well
+    if (
+      e.button !== 0 ||
+      e.altKey ||
+      e.ctrlKey ||
+      e.metaKey ||
+      e.shiftKey ||
+      e.defaultPrevented
+    ) {
+      redirect = false
+    }
+
+    if (props.target && props.target.toLowerCase() !== `_self`) {
+      redirect = false
+    }
+
+    if (tracking && window.ga) {
+      window.ga(`send`, `event`, {
+        eventCategory: `Outbound Link`,
+        eventAction: `click`,
+        eventLabel: `${tracking} - ${props.to || props.href}`,
+        transport: redirect ? `beacon` : ``,
```

Comment:
why wouldn't you use `beacon` for everything?

Reply:
As per https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits and https://caniuse.com/#feat=beacon I'm in the understanding that it's currently best to let the lib pick the transport - except for the case when we are redirecring and a plain old `fetch` is expected not to work.

Reply:
> By default, analytics.js picks the HTTP method and transport mechanism with which to optimally send hits. The three options are 'image' (using an Image object), 'xhr' (using an XMLHttpRequest object), or 'beacon' using the new navigator.sendBeacon method.

sweet! thought if you set it to beacon GA would use beacon and fallback but it's not super clear so 👍